crawl-ref/source/skills.cc

```diff
@@ -2092,6 +2092,15 @@ string skill_title_by_rank(skill_type best_skill, uint8_t skill_rank,
         default:
             break;
         }
+        // Make sure the traitor induced title overrides under penance
+        if (you.attribute[ATTR_TRAITOR] > 0)
+            {
+                for (god_iterator it; it; ++it)
+                {
+                    if (player_under_penance(*it))
+                        result = god_title(*it, species, piety);
+                }
+            }
```

Comment:
This will choose semi-arbitrarily between traitor titles if the player has multiple penances; in particular this can get you a traitor title by abandoning one god to get ATTR_TRAITOR to 1, then slightly upsetting your current god to pass the player_under_penance. See comment on 755-760 of religion.cc